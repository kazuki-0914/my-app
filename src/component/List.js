import React, { Component } from 'react';

// const items = [{
//     id: 1,
//     userId: 1,
//     title: 'delectus aut autem',
//     completed: 'false'
// },
// {
//     id: 12,
//     userId: 12,
//     title: 'delectus aut autem2',
//     completed: 'false'
// }]



export default class List extends Component {
    constructor(props) {
        super(props);
        this.state = {
            records: [],
        };
    }
    componentDidMount = () => {
        // const records = [];
        // items.forEach((item) => {
        //     records.push(<tr key={item.id}>
        //         <td>{item.id}</td>
        //         <td>{item.userId}</td>
        //         <td>{item.title}</td>
        //         <td>{item.completed}</td>
        //     </tr>);
        // });
        // console.log(`records`, records);
        // this.setState({
        //     records: records
        // });


        fetch('https://jsonplaceholder.typicode.com/todos')
            .then(response => response.json())
            .then(json => {
                const records = [];
                json.forEach((item) => {
                    records.push(<tr key={item.id}>
                        <td>{item.id}</td>
                        <td>{item.userId}</td>
                        <td>{item.title}</td>
                        <td>{item.completed}</td>
                    </tr>);
                });
                // console.log(`records`, records);
                this.setState({
                    records: records
                });
            })
    }

    render() {
        return (
            <div>

                <table border="1">
                <thead>
                    <tr>
                        <th>id</th>
                        <th>userId</th>
                        <th>title</th>
                        <th>completed</th>
                    </tr>
                   </thead>
                   <tbody>
                    {/* 
                    <tr>
                        <td>1</td>
                        <td>1</td>
                        <td>delectus aut autem</td>
                        <td>false</td>
                    </tr>
                     */}
                    {this.state.records}
                    </tbody>
                </table>
            </div>
        );
    }
}
