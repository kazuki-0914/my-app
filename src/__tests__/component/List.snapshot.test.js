'use strict';
import {render, fireEvent, waitFor, screen} from '@testing-library/react'
import List from '../../component/List';
import renderer from 'react-test-renderer';
import {rest} from 'msw'
import {setupServer} from 'msw/node'
// TODO 工事中
// import { render, unmountComponentAtNode } from "react-dom";
// let container = null;
// beforeEach(() => {
//   // setup a DOM element as a render target
//   container = document.createElement("div");
//   document.body.appendChild(container);
// });

// afterEach(() => {
//   // cleanup on exiting
//   unmountComponentAtNode(container);
//   container.remove();
//   container = null;
// });
// beforeAll(() =>  render(<Todo />))
// beforeEach(() =>  render(<List />))
// afterEach(() => server.resetHandlers())
// afterAll(() => server.close())
    const items = [{
      id: 1,
      userId: 1,
      title: 'delectus aut autem',
      completed: 'false'
  },
  {
      id: 12,
      userId: 12,
      title: 'delectus aut autem2',
      completed: 'false'
  }]
const server = setupServer(
  rest.get('https://jsonplaceholder.typicode.com/todos', (req, res, ctx) => {
    return res(ctx.json(items))
  }),
)

beforeAll(() => server.listen())
afterEach(() => server.resetHandlers())
afterAll(() => server.close())
// 初期表示
// test('初期表示', async () => {
//   render(<List />);
//   expect(screen.getByText('id')).toBeVisible()
// });
// 初期表示
test.skip('表示', async () => {
  render(<List />);
  console.log(`screen`,screen.debug());
  // act(() => {
  //   /* 状態を更新するイベントを発生させます */
  //   const items = [{
  //     id: 1,
  //     userId: 1,
  //     title: 'delectus aut autem',
  //     completed: 'false'
  // },
  // {
  //     id: 12,
  //     userId: 12,
  //     title: 'delectus aut autem2',
  //     completed: 'false'
  // }]
  //   jest.spyOn(global, "fetch").mockImplementation(() =>
  //     Promise.resolve({
  //       json: () => Promise.resolve(items)
  //     })
  //   );
  // });

  // const component = renderer.create(
  //   <List />,container
  // )
  // expect(component.toJSON()).toMatchSnapshot();

    // Use the asynchronous version of act to apply resolved promises
    // await act(async () => {
      // render(<List/>,container);
    // });
    // console.log(`screen`,container.textContent);
    // console.log(`container`,container.render());
  // expect(screen.getByText('delectus aut autem')).toBeVisible()
  // expect(container.getByText('delectus aut autem')).toBeVisible();
  // remove the mock to ensure tests are completely isolated
  // global.fetch.mockRestore();
});