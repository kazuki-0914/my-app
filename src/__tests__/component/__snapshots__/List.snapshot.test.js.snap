// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`表示 1`] = `
<div>
  <table
    border="1"
  >
    <thead>
      <tr>
        <th>
          id
        </th>
        <th>
          userId
        </th>
        <th>
          title
        </th>
        <th>
          completed
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
